actors/builtin/miner/miner_state.go

```diff
@@ -33,6 +34,9 @@ type State struct {
 	// Sectors that have been pre-committed but not yet proven.
 	PreCommittedSectors cid.Cid // Map, HAMT[SectorNumber]SectorPreCommitOnChainInfo
 
+	// Allocated sector IDs. Sector IDs can never be reused once allocated.
+	AllocatedSectors cid.Cid // BitField
```

Comment:
I've put this behind a CID as it _could_ get large.

Q: We can save loading a few bytes every time by grouping this with `PreCommittedSectors`. That is, we can define:

```go
type PreCommitInfo {
    PreCommittedSectors cid.Cid // Map
    AllocatedSectors BitField (or behind a CID to avoid loading on prove-commit?)
}
```

And only load this when we care about precommits.

But this is probably a low priority question we can punt.

Reply:
My plan is to punt on this for now.